### PR TITLE
Move to the Dev15 SDK for building VSIX

### PR DESCRIPTION
**Pull request by jaredpar**
This change has two interesting effects on our code base:

- Standardizes our Dev15 SDK references to Preview 5.  Previously it was a mix of Preview 4 and a build between Preview 4 and Preview 5.
- Moves all VSIX SDK tooling to Dev15 RC2 SDK.  

This change will break our ability to deploy VSIX on Dev14.  That is okay though as we are moving towards Dev15 for our primary development.  Building on Dev14 will still be supported, we will just lose the ability to F5 the changes. 

This doesn't fully move us to VSIX V3.  That will come later.  This is a necessary middle step to unblock other work.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @Pilchie, @dotnet/roslyn-infrastructure

File: build/config/RepoUtilData.json [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-        "Microsoft.VisualStudio.Text.Logic": [ "14.3.25407", "15.0.25604-Preview4" ],
-        "Microsoft.VisualStudio.Text.Data": [ "14.3.25407", "15.0.25604-Preview4" ],
-        "Microsoft.VisualStudio.Text.UI": "14.3.25407",
+        "Microsoft.VisualStudio.Editor": [ "14.3.25407", "15.0.25726-Preview5" ],
+        "Microsoft.VisualStudio.Imaging": [ "14.3.25407", "15.0.25726-Preview5" ],
+        "Microsoft.VisualStudio.ImageCatalog": [ "14.3.25407", "15.0.25726-Preview5" ],
+        "Microsoft.VisualStudio.Text.UI.Wpf": [ "14.3.25407", "15.0.25726-Preview5" ],
+        "Microsoft.VisualStudio.Text.Logic": [ "14.3.25407", "15.0.25726-Preview5" ],
+        "Microsoft.VisualStudio.Text.Data": [ "14.3.25407", "15.0.25726-Preview5" ],
+        "Microsoft.VisualStudio.Text.UI": "14.3.25407"
```
Comment (Pilchie) [inline build/config/RepoUtilData.json]: Missing `,`

File: build/Targets/Imports.targets [hunk 2]
```diff
@@ -290,30 +290,6 @@
...
-  <!-- Currently, the Willow installer can't install vsixes that have a <Prerequisites> element
-       but the Dev15 RC SDK packages can NOT install vsixes without them as part of the build.
-       To work around this conflict, we include them in our manifest files, but if we're building
-       with something older than Dev15 RC (as we do for official builds that we insert into setup)
-       we'll run a transform to strip out the <Prerequisites> element. -->
-  <Target Name="RewriteVsixManifestOnOfficialBuild"
-          BeforeTargets="FindSourceVsixManifest"
-          Condition="'$(IsDev15RC)' != 'true'">
-    <MakeDir Directories="$(IntermediateOutputPath)" />
-    <XslTransformation XslInputPath="$(MSBuildThisFileDirectory)TransformVsixManifest.xslt"
```
Comment (Pilchie) [inline build/Targets/Imports.targets]: Can you delete this `xslt` file too? Please?!

File: Restore.cmd [hunk 3]
```diff
@@ -41,8 +41,8 @@ call "%NugetExe%" restore "%RoslynRoot%build\ToolsetPackages\dev14.project.json"
 echo Restoring packages: Toolsets (Dev15 VS SDK build tools)
 call "%NugetExe%" restore "%RoslynRoot%build\ToolsetPackages\dev15.project.json" %NuGetAdditionalCommandLineArgs% || goto :RestoreFailed
 
-echo Restoring packages: Toolsets (Dev15 VS SDK 'Willow' build tools)
-call "%NugetExe%" restore "%RoslynRoot%build\ToolsetPackages\dev15Willow.project.json" %NuGetAdditionalCommandLineArgs% || goto :RestoreFailed
+echo Restoring packages: Toolsets (Dev15 VS SDK Preview 5build tools)
+call "%NugetExe%" restore "%RoslynRoot%build\ToolsetPackages\dev15p5.project.json" %NuGetAdditionalCommandLineArgs% || goto :RestoreFailed
```
Comment (Pilchie) [inline Restore.cmd]: Any reason to keep this (and the one above that is for P3) around if we're only ever going to build with Dev14 or Dev15 RC?

**Comment by jaredpar** [comment]
@Pilchie i'm scared to run VSI suites ...

**Comment by jaredpar** [comment]
test vsi please

File: build/Targets/Imports.targets (hunk 2 above)
Comment (jaredpar) [inline build/Targets/Imports.targets]: With pleasure

File: Restore.cmd (hunk 3 above)
Comment (jaredpar) [inline Restore.cmd]: Yeah those can be deleted too.

**Comment by jaredpar** [comment]
retest vsi please

**Comment by jaredpar** [comment]
retest windows_release_unit64_prtest please
retest windows_debug_unit64_prtest please
retest perf_correctness_prtest please

All failed due to Jenkins I/O issues.

**Comment by jaredpar** [comment]
retest windows_vsi_p0_open_prtest please 

Another Jenkins I/O error

**Comment by jaredpar** [comment]
retest this please

**Comment by jaredpar** [comment]
test vsi please

**Comment by jaredpar** [comment]
test vsi please

**Comment by jaredpar** [comment]
retest windows_vsi_p3_open_prtest please

**Comment by jaredpar** [comment]
retest windows_vsi_p3_open_prtest please

bad locked screen

**Pull request merged**